### PR TITLE
Implement GetFriendGamePlayed

### PR DESCRIPTION
**Pull request by Kruithne**
Implements the [GetFriendGamePlayed](https://partner.steamgames.com/doc/api/ISteamFriends#GetFriendGamePlayed) function, retrieving information regarding a given friend.

### CODE REVIEW

File: src/api/steam_api_friends.cc [hunk 1]
```diff
@@ -327,6 +327,38 @@ NAN_METHOD(ClearRichPresence) {
...
+
+  std::string steam_id_str(*(Nan::Utf8String(info[0])));
+  CSteamID steam_id(utils::strToUint64(steam_id_str));
+
+  if (!steam_id.IsValid()) {
+    THROW_BAD_ARGS("Steam ID is invalid");
+  }
+
+  FriendGameInfo_t* friendInfo = new FriendGameInfo_t();
+  SteamFriends()->GetFriendGamePlayed(steam_id, friendInfo);
```
Comment (hokein) [inline src/api/steam_api_friends.cc:344]: we should check the return value is `true` or `false`.

if false, we probably need to return a null object.

Comment (Kruithne) [reply src/api/steam_api_friends.cc]: Good catch. I couldn't see a means of returning `null`, but keeping in line with the rest of the API, returning `undefined` seemed fitting. Updated the PR.

File: src/api/steam_api_friends.cc [hunk 2]
```diff
@@ -327,6 +327,40 @@ NAN_METHOD(ClearRichPresence) {
...
+
+  std::string steam_id_str(*(Nan::Utf8String(info[0])));
+  CSteamID steam_id(utils::strToUint64(steam_id_str));
+
+  if (!steam_id.IsValid()) {
+    THROW_BAD_ARGS("Steam ID is invalid");
+  }
+
+  FriendGameInfo_t* friendInfo = new FriendGameInfo_t();
+  if (SteamFriends()->GetFriendGamePlayed(steam_id, friendInfo)) {
```
Comment (hokein) [inline src/api/steam_api_friends.cc:344]: nit: using early return can improve the code readability.

```
if (!...)
  ...
// propogate FriendInfo.
```

File: src/api/steam_api_friends.cc [hunk 3]
```diff
@@ -327,6 +327,40 @@ NAN_METHOD(ClearRichPresence) {
...
+  }
+
+  std::string steam_id_str(*(Nan::Utf8String(info[0])));
+  CSteamID steam_id(utils::strToUint64(steam_id_str));
+
+  if (!steam_id.IsValid()) {
+    THROW_BAD_ARGS("Steam ID is invalid");
+  }
+
+  FriendGameInfo_t* friendInfo = new FriendGameInfo_t();
```
Comment (hokein) [inline src/api/steam_api_friends.cc:343]: this looks like a memory leak, why not using a local variable?

File: docs/friends.md [hunk 4]
```diff
@@ -340,6 +340,12 @@ Mark a target user as 'played with'.
 
 * `steamIDUserPlayedWith` String: a 64-bits steam ID of other user that we have played with.
 
+### greenworks.getFriendGamePlayed(steamIDFriend)
+
+* `steamIDFriend` String: a 64-bit Steam ID.
+
+Returns an object containing [FriendGameInfo_t](https://partner.steamgames.com/doc/api/ISteamFriends#FriendGameInfo_t) values.
```
Comment (hokein) [inline docs/friends.md:347]: update the doc about the `undefined` case.

File: src/api/steam_api_friends.cc (hunk 3 above)
Comment (Kruithne) [reply src/api/steam_api_friends.cc:343]: I'm not familiar enough with the syntax here; first time using C++. Isn't `friendInfo` local to the scope? Would preferred syntax be `FriendGameInfo_t* friendInfo();` instead?

Comment (hokein) [reply src/api/steam_api_friends.cc:343]: Just  

```
FriendGameInfo_t friendInfo;
if (!SteamFriends()->GetFriendGamePlayed(steam_id, &friendInfo)) {
}
...
```

**Review by hokein — APPROVED** [review]
thanks.

**Pull request merged**